src/github-repo.ts

```diff
@@ -51,6 +52,36 @@ type Branch = {
   };
 };
 
+const setTimeoutAsync = promisify(setTimeout);
+
+async function waitFor(
+  fn: (...args: unknown[]) => Promise<unknown>,
+  timeout = 10000,
+  interval = 100,
+  message = 'Timeout exceeded for task'
+) {
+  let done = false;
+  const tid = setTimeout(() => {
+    done = true;
+  }, timeout);
+  try {
+    while (!done) {
+      try {
+        const res = await fn();
+        if (res) {
+          return res;
+        }
+        throw new Error('No result returned from wait fn');
```

Comment:
In a generic sense you'd probably want to differentiate between this error and errors thrown by fn(), (and at least log errors thrown by fn) but for this specific usecase it is probably OK.

Reply:
I sorta wanted this, but as always should've been more explicit about the intentions here, thanks for highlighting. How does 578248a90e1d0035e90dbeddc904d26aeb1bebb4 look to you?

Reply:
Looks good.